File: Python/bench_1_2.py
import numpy as np
from time import time

def bench_1_2():

    trials = 100
    rows = 1000
    cols = 1000

    times = []

    for i in range(trials):
    
        M = np.random.randint(0,10000, size=(rows,cols))
        MtM = np.matmul(np.transpose(M), M)

        start = time()
        L = np.linalg.cholesky(MtM)
        t = time()-start

        times.append(t)

    print 'Python - Benchmark 1.2: Average time = {} milliseconds'.format(np.mean(times)*1000)

File: Matlab/bench_1_2.m
function bench_1_2() 

trials = 100;
rows = 1000;
cols = 1000;

times = zeros(1,trials);

for i = 1:trials
    
   M = randi(10000, rows, cols);
   MtM = M'*M;
   
   tic();
   chol(MtM, 'lower');
   t = toc();
   
   times(i) = t;
   
end

fprintf('Matlab - Benchmark 1.2: Average time = %f milliseconds\n', mean(times)*1000);